source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -0,0 +1,206 @@
+#include "extensions/filters/http/cache/cache_filter.h"
+
+#include "envoy/registry/registry.h"
+
+#include "absl/memory/memory.h"
+#include "absl/strings/string_view.h"
+
+using absl::string_view;
```

Comment:
Let's not import absl methods into this file; just qualify absl::string_view at use, and we can use std::make_unique rather than WrapUnique. I don't believe we generally do that anywhere in Envoy other than the testing:: methods.

Reply:
- All using-declarations removed.

- These uses of WrapUnique can't be replaced by make_unique, as that would try to allocate a new (abstract) HeaderMap. I'd like to capture the HeaderMapPtrs into these std::functions, but std::function doesn't support move-only types (such as HeaderMapPtr), so this code instead capturing HeaderMap*s into the std::function, and shoves them back into HeaderMapPtrs at the earliest opportunity.